src/Init/System/IO.lean

```diff
@@ -291,6 +291,13 @@ MonadIOProcess.wait c
 end Proc
 -/
 
+structure Pipe :=
+mk' :: (input : FS.Handle)
+       (output : FS.Handle)
+
+@[extern "lean_io_mk_pipe"]
+constant Pipe.Prim.mk : IO Pipe := arbitrary _
```

Comment:
We can mark constructors with `extern` using the `attribute` command.
Example:
```lean
attribute [extern "lean_mk_thunk"] Thunk.mk
```

Reply:
I don't think that helps here because the constructor for `Pipe` takes two handles but we actually need those handles to be created on the C++ side.